common/src/main/java/io/pravega/common/util/SimpleDeque.java

```diff
@@ -0,0 +1,315 @@
+        return (T) this.items[this.head];
+    }
+
+    /**
+     * Gets a value indicating the number of items in the {@link SimpleDeque}.
+     *
+     * @return The size of the {@link SimpleDeque}.
+     */
+    public int size() {
+        final int size = this.tail - this.head;
+        return size < 0 ? size + this.items.length : size;
+    }
+
+    /**
+     * Gets a value indicating whether the {@link SimpleDeque} is empty or not.
+     * Consider using this instead of {@code {@link #size()} == 0} since this method does not need to compute anything.
+     *
+     * @return True if empty, false otherwise.
+     */
+    public boolean isEmpty() {
+        return this.head == this.tail;
+    }
+
+    /**
+     * Clears the entire {@link SimpleDeque}.
+     */
+    public void clear() {
+        Arrays.fill(this.items, null);
+        this.head = 0;
+        this.tail = 0;
+    }
+
+    private int increment(int index) {
+        index++;
+        return index == this.items.length ? 0 : index;
+    }
+
+    private void expand() {
+        int newCapacity = this.items.length;
+        if (this.items.length >= MAX_CAPACITY) {
+            // Can't allocate an array bigger than this.
+            throw new OutOfMemoryError("Unable to grow SimpleDequeue.");
+        }
+
+        if (newCapacity >= MAX_HALF) {
+            // We double capacity, so if we're already above half of max, then just set it to max.
+            newCapacity = MAX_CAPACITY;
+        } else {
+            // Double capacity.
+            newCapacity = newCapacity * 2;
+        }
+
+        final Object[] newItems = Arrays.copyOf(this.items, newCapacity);
+        boolean flipped = this.tail < this.head || (this.tail == this.head && newItems[this.head] != null);
+        if (flipped) {
+            final int diff = newCapacity - this.items.length;
+            System.arraycopy(newItems, this.head, newItems, this.head + diff, this.items.length - this.head);
+            int idx = this.head;
+            this.head += diff;
+            while (idx < this.head) {
+                newItems[idx] = null;
+                idx++;
+            }
+        }
+
+        this.items = newItems;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("Size = %s, Capacity = %s, Head = %s, Tail = %s", size(), this.items.length, this.head, this.tail);
+    }
+
+    //endregion
+
+    //region WrapQueue
+
+    @RequiredArgsConstructor
+    @NotThreadSafe
+    private static class WrapQueue<T> extends AbstractCollection<T> implements Queue<T> {
+        private final Object[] items;
+        private int nextIndex = 0;
+
+        //region Queue Implementation
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public T poll() {
+            if (this.nextIndex >= this.items.length) {
+                return null;
+            }
+
+            return (T) this.items[this.nextIndex++];
+        }
+
+        @Override
+        public T element() {
```

Comment:
Per the documentation here:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Queue.html
Element is a variation on `peek` not on `poll`.

Reply:
Fixed